### DISCUSSION TITLE
How do i input multiple strings or numbers seperately?

### DISCUSSION THREAD

**Discussion by 42sungwook** [community/community#21407] [Other Feature Feedback, Questions, & Ideas]
<h2>
<a name="a-b-inputsplit-c-ab-printc-1" class="anchor" href="#a-b-inputsplit-c-ab-printc-1"></a>a, b = input().split()<br>
c = a+b<br>
print(c)</h2>
<p>this gives me the answer i want however</p>
<h2>
<a name="a-input-b-input-c-ab-printc-2" class="anchor" href="#a-input-b-input-c-ab-printc-2"></a>a = input()<br>
b = input()<br>
c = a+b<br>
print(c)</h2>
<p>does not give me the answer<br>
this only prints out a<br>
how can i input multiple times and still make it work like the first example?</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hi <a class="mention" href="/wookgyeo">@WookGyeo</a> and welcome to the forum!</p>
<p>Here’s your code formatted in a code block for readability:</p>
<pre><code class="lang-python"># answer you want
a, b = input().split()
c = a + b
print(c)

# does not give answer, only prints out a
a = input()
b = input()
c = a + b
print(c)
</code></pre>
<p>It can help to be a bit more explicit about what you actually want. Since I don’t know what your goal is, it’s hard to help you out. I can explain what the code does in both cases, so you’re hopefully able to determine for yourself what is going wrong.</p>
<ul>
<li>The first piece of code takes input from the command line, and uses <code>split()</code> to split up the input string. You can provide an argument to <code>split</code> to tell Python where to split the string on, but if you don’t (as you did here), it’ll split on any whitespace. By using <code>a, b</code> to <em>unpack</em> the resulting list, you tell Python to put the first item of the list into <code>a</code> and the second item of the list into <code>b</code>. Note this throws an error if your list is not exactly two items long (if I input <code>test</code> into your program, it’ll throw an error). Assuming no error was thrown, you then glue the two strings together and store that into <code>c</code>. Hence, if I input <code>te st</code> into the program, <code>a</code> will be <code>te</code>, <code>b</code> will be <code>st</code> and <code>c</code>  will be <code>test</code>, which will be printed out.</li>
<li>In the second program, you take two separate inputs. You then glue these together and print the result. If I input <code>test</code> the first time, then input <code>ing</code> the second time, <code>a</code> will be <code>test</code>, <code>b</code> will be <code>ing</code>, and <code>c</code> will be <code>testing</code>, which will be printed.</li>
</ul>
